modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ServiceTests.java

```diff
@@ -217,4 +221,19 @@ public void testGetClientRegionFallbackToUsEast1() {
             );
         }
     }
+
+    public void testEndpointOverrideSchemeDefaultsToHttpsWhenNotSpecified() {
+        final S3Service s3Service = new S3Service(
+            mock(Environment.class),
+            Settings.EMPTY,
+            mock(ResourceWatcherService.class),
+            () -> Region.of("es-test-region")
+        );
+        String servername = randomIdentifier() + ".ignore";
```

Comment:
`endpointWithoutScheme`?

`servername` seems like surprising terminology, but maybe I'm missing the context.

Reply:
Fixed in b84cf5ebdbcb273e63130defaeb8329a3da9e413